lib/ssh_scan/rule_engine.rb

```diff
@@ -0,0 +1,94 @@
+module SSHScan
+  class RuleEngine
+    # Simple one rule matching.
+    def self.matchSingleRule(key, value, result)
```

Comment:
This definitely looks promising. Opened up #319 for this. I think we should pursue using YARD in a different PR.